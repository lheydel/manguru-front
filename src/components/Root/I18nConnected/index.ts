export * from './I18nConnected';
