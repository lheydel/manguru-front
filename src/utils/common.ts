import { AxiosRequestConfig } from 'axios';

export const requestConfig: AxiosRequestConfig = {
    withCredentials: true,
    headers: {
        'Content-Type': 'application/json'
    }
};
