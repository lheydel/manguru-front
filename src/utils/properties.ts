export enum Language {
    DEFAULT = 'en',
    EN = 'en',
    FR = 'fr'
}
