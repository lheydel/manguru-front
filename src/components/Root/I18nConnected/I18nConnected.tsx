import React, { ReactNode } from 'react';
import { I18nProvider } from '@lingui/react';
import { Language } from '../../../utils/properties';
import { ApplicationState } from '../../../services/common/app.state';
import { connect } from 'react-redux';
import i18nService from '../../../services/i18n/service/i18n.service';

export interface I18nConnectedProps {
  children: ReactNode;
  language: Language;
}

/**
 * Connect I18nProvider to the redux store
 */
export const RawI18nConnected: React.FC<I18nConnectedProps> = (props) => {
    const { i18n } = i18nService;
    const { language, children } = props;
    i18n.activate(language);

    return (
        <I18nProvider language={language} i18n={i18n}>
            {children}
        </I18nProvider>
    );
};

const mapStateToProps = (state: ApplicationState) => ({
    language: state.i18n.language
});

export const I18nConnected = connect(mapStateToProps)(RawI18nConnected);
