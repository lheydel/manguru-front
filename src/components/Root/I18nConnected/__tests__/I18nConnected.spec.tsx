import React from 'react';
import { Provider } from 'react-redux';
import I18nConnected, { RawI18nConnected } from '..';
import { shallow, mount } from 'enzyme';
import createMockStore from 'redux-mock-store';
import { initialAppState } from '../../../../services/common/app.state';
import toJson from 'enzyme-to-json';
import { I18nProvider } from '@lingui/react';
import { Language } from '../../../../utils/properties';

describe('I18nConnected component', () => {
    const child = <button />;

    it('should render itself and children when connected', () => {
        const wrapper = shallow(
            <Provider store={createMockStore()(initialAppState)}>
                <I18nConnected>{child}</I18nConnected>
            </Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should use default language', () => {
        const wrapper = mount(
            <Provider store={createMockStore()(initialAppState)}>
                <I18nConnected>{child}</I18nConnected>
            </Provider>
        );
        expect(wrapper.find(I18nProvider).props().language).toBe(initialAppState.i18n.language);
    });
});
