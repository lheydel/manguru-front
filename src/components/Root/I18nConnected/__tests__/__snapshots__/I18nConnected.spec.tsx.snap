// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I18nConnected component should render itself and children 1`] = `
<I18nProvider
  defaultRender={null}
  i18n={
    I18n {
      "_activeLanguageData": Object {
        "plurals": [Function],
      },
      "_activeMessages": Object {
        "Hello <0>{name}</0>! How are <1>you</1>?": [Function],
        "This is a test": "This is a test",
        "Today: {0,date}": [Function],
        "this is a test": "this is a test",
        "{0, plural, =0 {There is no banana :(} =666 {MOUHAHAHAHA I got my # bananas!!} one {Well, at leat we have one banana...} other {Look at all those bananas! We have at least # of them!!!!}}": [Function],
      },
      "_catalogs": Object {
        "en": Object {
          "languageData": Object {
            "plurals": [Function],
          },
          "messages": Object {
            "Hello <0>{name}</0>! How are <1>you</1>?": [Function],
            "This is a test": "This is a test",
            "Today: {0,date}": [Function],
            "this is a test": "this is a test",
            "{0, plural, =0 {There is no banana :(} =666 {MOUHAHAHAHA I got my # bananas!!} one {Well, at leat we have one banana...} other {Look at all those bananas! We have at least # of them!!!!}}": [Function],
          },
        },
        "fr": Object {
          "languageData": Object {
            "plurals": [Function],
          },
          "messages": Object {
            "Hello <0>{name}</0>! How are <1>you</1>?": [Function],
            "This is a test": "Ziz iz eu teste",
            "Today: {0,date}": [Function],
            "this is a test": "Ziz iz eu teste",
            "{0, plural, =0 {There is no banana :(} =666 {MOUHAHAHAHA I got my # bananas!!} one {Well, at leat we have one banana...} other {Look at all those bananas! We have at least # of them!!!!}}": [Function],
          },
        },
      },
      "_dev": Object {
        "compile": [Function],
        "loadLanguageData": [Function],
      },
      "_language": "en",
      "_locales": undefined,
      "plural": [Function],
      "select": [Function],
      "selectOrdinal": [Function],
      "t": [Function],
    }
  }
  language="en"
>
  <button />
</I18nProvider>
`;

exports[`I18nConnected component should render itself and children when connected 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Connect(RawI18nConnected)>
    <button />
  </Connect(RawI18nConnected)>
</ContextProvider>
`;
